### PR TITLE
fix(distributor): Parsing of url forces scheme to http or https

### LINKED ISSUE #7577: Distributor crashes for a malformed Keptn API endpoint
**Environment**

* Client OS (e.g., Linux, macOS, Windows): Linux
* Keptn Version (`keptn version`): 0.12.6, 0.13.4, 0.14.1
* Kubernetes Cloud Provider (e.g., GKE, AKS): k3d, GKE
* Kubernetes version (`kubectl version`): Client: v1.23.5; Server: v1.22.6+k3s1

**Affected Component**

* [ ] Approval
* [ ] Bridge
* [ ] CLI
* [ ] Datastore
* [ ] Docs
* [x] Distributor / Event Broker
* [ ] Helm
* [ ] Istio
* [ ] Jmeter
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] REST API
* [ ] Webhook

**Describe the bug**

The distributor crashes with the following output if a malformed Keptn API endpoint is specified:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x82a3d4]

goroutine 1 [running]:
main.getUniformHandlers(0x916f9f, 0x4, 0xc00002a050, 0x2d)
	/go/src/github.com/keptn/keptn/distributor/cmd/main.go:157 +0xd4
main._main(0xc000028033, 0x7, 0xc00002a050, 0x2d, 0x1f91, 0x853226, 0x1, 0x86b6fb, 0x2, 0x8778c5, ...)
	/go/src/github.com/keptn/keptn/distributor/cmd/main.go:52 +0x225
main.main()
	/go/src/github.com/keptn/keptn/distributor/cmd/main.go:40 +0x118

goroutine 17 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:168 +0xa5
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x25
created by os/signal.Notify.func1.1
	/usr/local/go/src/os/signal/signal.go:151 +0x45

goroutine 33 [runnable]:
main.createExecutionContext.func1(0xc000080b40, 0xc0003fe000)
	/go/src/github.com/keptn/keptn/distributor/cmd/main.go:106
created by main.createExecutionContext
	/go/src/github.com/keptn/keptn/distributor/cmd/main.go:106 +0xfb
```	

**To Reproduce**

1. Set the environment variable `KEPTN_API_ENDPOINT` in a deployment.yaml for the distributor to an malformed URL. E.g.: `KEPTN_API_ENDPOINT=":///api"`
2. Deploy the distributor 
3. Display the log for the container

**Expected behavior**

The distributor displays a error message that is easier to understand than the currently displayed stack trace.

**Current behavior**

The distributor crashes with a panic (`panic: runtime error: invalid memory address or nil pointer dereference`)

### PR DESCRIPTION
**Pull request by RealAnna**
### Related Issues

Fixes #7577

### How to test

set a distributor to an image based on this PR, add the KEPTN_API_ENDPOINT to be a malformed address, check distributor output
image

image

### CODE REVIEW

File: distributor/cmd/main.go [hunk 1]
```diff
@@ -173,8 +174,18 @@ func createKeptnAPI(httpClient *http.Client, env config.EnvConfig) (keptnapi.Kep
...
+
+		if env.KeptnAPIEndpoint != "" && parsed.Scheme == "" {
+			return nil, fmt.Errorf("invalid scheme for keptn endpoint, %s is not http or https", env.KeptnAPIEndpoint)
+		}
+
+		if strings.HasPrefix(parsed.Scheme, "http") {
+			// if no value is assigned to the endpoint than we keep the default scheme
+			// otherwise we override
 			scheme = parsed.Scheme
 		}
```
Comment (thisthat) [inline distributor/cmd/main.go:190]: I would expect an error in the `else` case

Comment (RealAnna) [reply distributor/cmd/main.go:190]: checked for other protocols in the previous branch since an else would also give error for the default empty endpoint

**Review by thisthat — APPROVED** [review]

**Review by thisthat — CHANGES_REQUESTED** [review]
Please, add also a test for this 😇

**Review by thisthat — APPROVED** [review]
Thank you

**Pull request merged**